WebAppKeyVault/WebAppKeyVault/Controllers/HomeController.cs

```diff
@@ -1,34 +1,30 @@
-﻿using System;
+﻿using Azure;
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+using System;
+using System.Threading.Tasks;
 using System.Web.Mvc;
-using Microsoft.Azure.KeyVault;
-using Microsoft.Azure.Services.AppAuthentication;
 
 namespace WebAppKeyVault.Controllers
 {
     public class HomeController : Controller
     {
-        public async System.Threading.Tasks.Task<ActionResult> Index()
+        public async Task<ActionResult> Index()
         {
-            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
-
             try
             {
-                var keyVaultClient = new KeyVaultClient(
-                    new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
+                string uri = Environment.GetEnvironmentVariable("KEY_VAULT_URI");
+                var client = new SecretClient(new Uri(uri), new DefaultAzureCredential());
```

Comment:
Use explicit local types for samples - no `var`.

Reply:
Resolved